src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -0,0 +1,429 @@
+            .addValue("workspace_id", workspaceId.toString())
+            .addValue("offset", offset)
+            .addValue("limit", limit);
+    List<DbResource> dbResourceList = jdbcTemplate.query(sql, params, DB_RESOURCE_ROW_MAPPER);
+
+    return dbResourceList.stream()
+        .map(this::constructResource)
+        .map(ReferenceResource.class::cast)
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Retrieve a resource by ID
+   *
+   * @param workspaceId identifier of workspace for the lookup
+   * @param resourceId identifer of the resource for the lookup
+   * @return WsmResource object
+   */
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      isolation = Isolation.SERIALIZABLE,
+      readOnly = true)
+  public WsmResource getResource(UUID workspaceId, UUID resourceId) {
+    return getResourceWithId(workspaceId, resourceId);
+  }
+
+  /**
+   * Retrieve a data reference by name. Names are unique per workspace.
+   *
+   * @param workspaceId identifier of workspace for the lookup
+   * @param name name of the resource
+   * @return WsmResource object
+   */
+  @Transactional(
+      propagation = Propagation.REQUIRED,
+      isolation = Isolation.SERIALIZABLE,
+      readOnly = true)
+  public WsmResource getResourceByName(UUID workspaceId, String name) {
+    return getResourceWithName(workspaceId, name);
+  }
+
+  // -- Reference Methods -- //
+
+  /**
+   * Create a reference in the database We do creates in flights where the same create is issues
+   * more than once.
+   *
+   * @param resource a filled in reference resource
+   * @throws DuplicateResourceException on a duplicate resource_id or (workspace_id, name)
+   */
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public void createReferenceResource(ReferenceResource resource)
+      throws DuplicateResourceException {
+    storeResource(
+        resource.getWorkspaceId(),
+        resource.getResourceId(),
+        resource.getName(),
+        resource.getDescription(),
+        REFERENCE,
+        resource.getResourceType(),
+        resource.getCloningInstructions(),
+        resource.attributesToJson(),
+        Optional.empty(),
+        Optional.empty(),
+        Optional.empty());
+  }
+
+  // -- Controlled Resource Methods -- //
+
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public boolean updateReferenceResource(
+      UUID workspaceId, UUID referenceId, String name, String description) {
+    if (name == null && description == null) {
+      throw new InvalidDaoRequestException("Must specify name or description to update.");
+    }
+
+    var params = new MapSqlParameterSource();
+
+    if (name != null) {
+      params.addValue("name", name);
+    }
+
+    if (description != null) {
+      params.addValue("description", description);
+    }
+
+    return updateResource(workspaceId, referenceId, params);
+  }
+
+  // -- Private Methods -- //
```

Comment:
This divider needs to be moved down, or just deleted. I don't see us maintaining things like this.